### PR TITLE
Add parity problem example

### PR DESCRIPTION
**Pull request by stas-sl**
I added this example as a follow-up of discussion here https://groups.google.com/forum/#!topic/blocks-users/Ku2fRCmcXaY

### CODE REVIEW

File: parity_problem/__init__.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    train_monitor = TrainingDataMonitoring(variables=[cost], prefix="train",
+                                           after_epoch=True)
+
+    main_loop = MainLoop(algorithm, stream_train,
+                         extensions=[test_monitor, train_monitor,
+                                     FinishAfter(after_n_epochs=num_epochs),
+                                     Printing(), ProgressBar()])
+    main_loop.run()
+
+    print 'Learned weights:'
```
Comment (rizar) [inline parity_problem/__init__.py]: Could you please make it Python 3 compatible?

**Comment by rizar** [comment]
This is awesome! Please add a test (we just want to be sure that the script is always runnable) and I will be very happy to accept this PR.

**Comment by stas-sl** [comment]
Done!

**Comment by rizar** [comment]
Great, thanks for this contribution! We really lacked a simple and compact example of RNN.

And of course, I am excited to accept the first PR from Belarus. :)

**Comment by stas-sl** [comment]
Ha-ha :) Go, Belarus!

Hopefully I will manage to understand your more advanced examples soon.

**Pull request merged**